generator/sbpg/targets/resources/sbp_messages_mod.rs

```diff
@@ -57,3 +58,55 @@ impl SBP {
         }
     }
 }
+
+impl SBPMessage for SBP {
+    fn get_message_type(&self) -> u16 {
+        match self {
+            ((*- for m in msgs *))
+            SBP::(((m.identifier|camel_case)))( msg ) => msg.get_message_type(),
+            ((*- endfor *))
+            SBP::Unknown{msg_id, ..} => *msg_id,
+        }
+    }
+
+    fn get_sender_id(&self) -> Option<u16> {
+        match self {
+            ((*- for m in msgs *))
+            SBP::(((m.identifier|camel_case)))( msg ) => msg.get_sender_id(),
+            ((*- endfor *))
+            SBP::Unknown{sender_id, ..} => Some(*sender_id),
+        }
+    }
+
+    fn set_sender_id(&mut self, new_id: u16) {
+        match self {
+            ((*- for m in msgs *))
+            SBP::(((m.identifier|camel_case)))( msg ) => msg.set_sender_id(new_id),
+            ((*- endfor *))
+            SBP::Unknown{sender_id, ..} => *sender_id = new_id,
+        }
+    }
+
+    fn into_frame(self) -> Vec<u8> {
+        match self {
+            ((*- for m in msgs *))
+            SBP::(((m.identifier|camel_case)))( msg ) => msg.into_frame(),
+            ((*- endfor *))
+            SBP::Unknown{msg_id, sender_id, payload} => {
+                let mut frame = Vec::new();
+                frame.push(0x55);
+                frame.extend(msg_id.into_sbp_bytes());
+                frame.extend(sender_id.into_sbp_bytes());
+                frame.push(payload.len() as u8);
+                frame.extend(payload);
+
+                let mut crc = crc16::State::<crc16::XMODEM>::new();
+                crc.update(&frame[1..]);
+                frame.extend(crc.get().into_sbp_bytes());
+
+                frame
```

Comment:
regardless of how we end up doing framing we need to check that the total size is less than the max payload size of SBP

Reply:
Done! I ended up added a whole separate error and result type just for the framing.